### PR TITLE
feat: add OpenAPI support

### PR DESCRIPTION
**Pull request by alanpoulain**

### CODE REVIEW

File: src/OpenApi/ClassGenerator.php [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+            $listOperation = $pathCollection->get ?? null;
+            $createOperation = $pathCollection->post ?? null;
+            $class->operations = [
+                'item' => array_merge(
+                    $showOperation ? ['get' => null] : [],
+                    $putOperation ? ['put' => null] : [],
+                    $patchOperation ? ['patch' => null] : [],
+                    $deleteOperation ? ['delete' => null] : [],
+                ),
+                'collection' => array_merge(
```
Comment (soyuka) [inline src/OpenApi/ClassGenerator.php:133]: we're trying to move away from item/collection in API Platform, is this still wanted?

File: src/OpenApi/PropertyGenerator/PropertyGenerator.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+declare(strict_types=1);
+
+namespace ApiPlatform\SchemaGenerator\OpenApi\PropertyGenerator;
+
+use ApiPlatform\SchemaGenerator\Model\Class_;
+use ApiPlatform\SchemaGenerator\Model\Property;
+use ApiPlatform\SchemaGenerator\OpenApi\Model\Property as OpenApiProperty;
+use ApiPlatform\SchemaGenerator\PropertyGenerator\PropertyGenerator as CommonPropertyGenerator;
+use ApiPlatform\SchemaGenerator\PropertyGenerator\PropertyGeneratorInterface;
+use cebe\openapi\spec\Schema;
```
Comment (soyuka) [inline src/OpenApi/PropertyGenerator/PropertyGenerator.php:21]: why not use the ApiPlatform Schema object?

File: src/Model/Property.php [hunk 3]
```diff
@@ -13,26 +13,29 @@
...
 namespace ApiPlatform\SchemaGenerator\Model;
 
-use EasyRdf\Resource as RdfResource;
 use Nette\PhpGenerator\ClassType;
 use Nette\PhpGenerator\Method;
 use Nette\PhpGenerator\PhpNamespace;
 use Nette\PhpGenerator\Property as NetteProperty;
 
-final class Property
+abstract class Property
```
[the remark this answers is not in the document]
Comment (alanpoulain) [reply src/Model/Property.php:21]: We don't want to depend on API Platform, but yes there are some similarities (but also differences).

File: src/OpenApi/ClassGenerator.php (hunk 1 above)
Comment (alanpoulain) [reply src/OpenApi/ClassGenerator.php:133]: Operations are still managed the old way for now. We can change this in the future.

File: src/OpenApi/PropertyGenerator/PropertyGenerator.php (hunk 2 above)
Comment (alanpoulain) [reply src/OpenApi/PropertyGenerator/PropertyGenerator.php:21]: We don't want to depend on API Platform. However if we make a separate library, we could replace this dependency in the future.

**Pull request merged**